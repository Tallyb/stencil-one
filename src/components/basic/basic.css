.nice {
    background: blue;
    color: honeydew;

}