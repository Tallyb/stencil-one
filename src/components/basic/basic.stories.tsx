import { h } from '@stencil/core';

export default {
  title: 'My Basic',
};

export const Default = () => (<my-basic first="Millie" middle="Bobby" last="Brown"></my-basic>);

