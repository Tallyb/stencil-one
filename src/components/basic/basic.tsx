import { Component, Prop, h, JSX } from '@stencil/core';

@Component({
  tag: 'my-basic',
  styleUrl: 'basic.css',
  shadow: true
})
export class MyBasic {

    @Prop() first: string;

    @Prop() last: string;
  
    render(): JSX.Element {
      return (
        <div>
          <p class="nice">My name is {this.last} {this.first}</p>
          <slot/>
        </div>
            
      );
    }
}