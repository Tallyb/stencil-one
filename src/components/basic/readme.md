# my-basic



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description | Type     | Default |
| -------- | --------- | ----------- | -------- | ------- |
| `first`  | `first`   |             | `string` | `''`    |
| `last`   | `last`    |             | `string` | `''`    |


## Dependencies

### Used by

 - [my-nested](../nested)

### Graph
```mermaid
graph TD;
  my-nested --> my-basic
  style my-basic fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
