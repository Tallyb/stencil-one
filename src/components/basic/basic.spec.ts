import { newSpecPage } from "@stencil/core/testing";

import { MyBasic } from './basic';

describe('basic', () => {

    const html = `
    <my-basic first="Julia" last="Roberts">
        <p slot class="cool">Cool!</p>
    </my-basic>
    `;
    const components = [MyBasic];

    it('should render with shadow dom', async () => {
        const page = await newSpecPage({
            html,
            components
        });
        expect(page.root.shadowRoot).toBeTruthy();
        expect(page.root.querySelector('.nice')).toBeFalsy();
        expect(page.root.shadowRoot.querySelector('.nice')).toBeTruthy();
        expect(page.root.querySelector('.cool')).toBeTruthy();
        expect(page.root).toMatchSnapshot();
    });

    it('should render without shadow dom', async () => {
        const page = await newSpecPage({
            html,
            components,
            supportsShadowDom: false
        });
        expect(page.root.shadowRoot).toBeFalsy();
        expect(page.root.querySelector('.nice')).toBeTruthy();
        expect(page.root.querySelector('.cool')).toBeTruthy();

    });

});