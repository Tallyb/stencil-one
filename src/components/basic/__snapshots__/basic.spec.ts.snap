// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic should render with shadow dom 1`] = `
<my-basic first="Julia" last="Roberts">
  <mock:shadow-root>
    <div>
      <p class="nice">
        My name is Roberts Julia
      </p>
      <slot></slot>
    </div>
  </mock:shadow-root>
  <p class="cool" slot="">
    Cool!
  </p>
</my-basic>
`;
